spiceypy/spiceypy.py

```diff
@@ -10985,8 +10993,16 @@ def scencd(sc, sclkch, MXPART=None):
     sc = ctypes.c_int(sc)
     sclkch = stypes.stringToCharP(sclkch)
     sclkdp = ctypes.c_double()
-    libspice.scencd_c(sc, sclkch, ctypes.byref(sclkdp))
-    return sclkdp.value
+    if hasattr(sclkch, "__iter__"):
```

Comment:
for this function and for the other 4, change to use new iterable test function to fix failed builds
```python
if stypes.isiterable(sclkch):
```

Reply:
what about the other 14 uses of `if hasattr`?

Reply:
I can fix all of them if you want, IF you let me remove those un-holy white-spaces at the end of lines and in empty lines as well? Deal? ;)

Reply:
btw, i also found functions where you accept both scalar and iterable, but return type in the docstring does not mention it, e.g. srfxpt . Do those docstrings also need fixing?

Reply:
another btw: as we want SpiceyPy to become PEP8 compliant in the future, right, we need to work on some stuff anyway. My editor removes them by default, that's why the white space was gone.

Quote from PEP8:

> Avoid trailing whitespace anywhere. Because it's usually invisible, it can be confusing: e.g. a backslash followed by a space and a newline does not count as a line continuation marker. Some editors don't preserve it and many projects (like CPython itself) have pre-commit hooks that reject it.

Reply:
i have updated all `if hasattr` to `stypes.isiterable` if they fit the pattern of looping over a potential iterable, otherwise do skalar type return. There were some cases with cells where that pattern didn't match, so i didn't touch those.